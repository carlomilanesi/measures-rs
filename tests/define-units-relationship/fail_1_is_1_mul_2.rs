measures::define_measure_types! {
    exact,
    []
}

measures::measurement_vector_property! { P1 }

pub struct U1;
impl MeasurementUnit for U1 {
    type Property = P1;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u1";
}

measures::measurement_vector_property! { P2 }

pub struct U2;
impl MeasurementUnit for U2 {
    type Property = P2;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u2";
}

measures::measurement_vector_property! { P3 }

pub struct U3;
impl MeasurementUnit for U3 {
    type Property = P3;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u3";
}

measures::define_units_relationship! { true false false, U1 1 == U2 1 * U3 2 }

fn main() {}
