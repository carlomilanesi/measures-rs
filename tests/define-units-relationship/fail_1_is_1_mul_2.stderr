error: no rules expected `2`
  --> tests/define-units-relationship/fail_1_is_1_mul_2.rs:23:27
   |
23 |         U1 1 == U2 1 * U3 2,
   |                           ^ no rules expected this token in macro call
   |
note: while trying to match `1`
  --> src/define_units_relationship.rs
   |
   |     { $exact:tt $with_approx:tt $with_correlation:tt, $unit1:ident 1 == $unit2:ident 1 * $unit3:ident 1 } => {
   |                                                                                                       ^
